lib/boba/enums_railtie.rb

```diff
@@ -0,0 +1,77 @@
+# typed: true
+# frozen_string_literal: true
+
+require "rails/railtie"
+
+module Boba
+  class EnumsRailtie < Rails::Railtie
+    railtie_name(:boba)
+
+    initializer("boba.add_enum_classes") do
+      ActiveSupport.on_load(:active_record) do
+        module ActiveRecordTypedEnum
+          extend T::Sig
+          extend T::Helpers
+
+          module ClassMethods
+            extend T::Sig
+            extend T::Helpers
+            include Kernel
+
+            abstract!
+
+            sig { abstract.returns(T::Hash[String, T::Hash[String, T.untyped]]) }
+            def defined_enums; end
+
+            sig { abstract.params(name: String, value: T.untyped).returns(T.untyped) }
+            def const_set(name, value)
+            end
+
+            sig { abstract.params(name: String, block: T.proc.params(args: T.untyped).void).returns(T.untyped) }
+            def define_method(name, &block)
+            end
+
+            sig { params(definitions: T::Hash[T.untyped, T.untyped]).returns(T.untyped) }
+            def enum(definitions)
+              result = super
+
+              # Add typed enum methods for each enum
+              definitions.each_key do |name|
+                enum_values = defined_enums.fetch(name.to_s)
```

Comment:
So `definitions` is what you are passing to the Rails `enum` method.

We do NOT want to actually use this for defining the `T::Enum` because Rails does some normalization on it.

So instead we are using the `defined_enums` method that exists on ActiveRecord models, and then specifically looking for the enum definition for the key we were passed on!

This way we can assume that everything in this array is stringified, and its in the format we expect!